### ISSUE TITLE
dcgm-exporter high cpu usage 

### ISSUE THREAD

**Issue by ysshaoxiao** [NVIDIA/gpu-monitoring-tools#133]
![image](https://user-images.githubusercontent.com/14806133/100055938-b21dcb80-2e5f-11eb-8c7c-327ecc50237c.png)

Why is CPU utilization high？

**Comment by itzsimpl** [comment]
I'm also seeing high cpu usage with latest image.
<img width="959" alt="image" src="https://user-images.githubusercontent.com/4784070/101288792-b26f7c80-37f8-11eb-9369-4c93d52053af.png">

**Comment by Moersity** [comment]
![image](https://user-images.githubusercontent.com/17212638/101593935-cd760280-3a2b-11eb-8091-0379200b0d2c.png)
me too
image:  nvidia/dcgm-exporter:2.0.13-2.1.1-ubuntu18.04

**Comment by jelmd** [comment]
Yes, 1 CPU core @ 15..20% for a couple of metrics seems pretty high (Intel Xeon Gold 5115 CPU @ 2.40GHz).
Also 71 MiB RSS for a gathering agent seems to be very high as well, even if it is a go application.

**Comment by jelmd** [comment]
FWIW: `/usr/bin/nvidia-smi dmon -s puct` uses ~ 5 MiB RSS and 1 CPU core @ 7.5 +- 2.5% on the same machines, only.

**Comment by vivisidea** [comment]
me too, any updates?  

![image](https://user-images.githubusercontent.com/652222/123586746-c4a61c00-d817-11eb-84db-1d38094146d7.png)

**Comment by jelmd** [comment]
try nvmex. It seems to be a little bit better, but the root cause seems to be the inefficient NVML.
